### PR TITLE
Add a caution note

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Added Kubernetes as Kyma installation prerequisite
- Added a caution note about deprecating Docker in Kubernetes

### CODE REVIEW

**Review by janmedrek — DISMISSED** [review]

**Review by kwiatekus — DISMISSED** [review]

**Comment by klaudiagrz** [comment]
Resolves this issue:
#12558

File: docs/02-get-started/README.md [hunk 1]
```diff
@@ -9,7 +9,10 @@ All the steps are performed in the `default` Namespace.
 
 ## Prerequisites
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (1.19 or greater)
+>**CAUTION:** As of version 1.20, [Kubernetes deprecated Docker](https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/) as a container runtime in favor of [containerd](https://containerd.io/). Due to a different way in which containerd handles certificate authorities, Kyma's built-in Docker registry does not work correctly on clusters running with a self-signed TLS certificate on top of Kubernetes installation where containerd is used as a container runtime. If that is your case, either upgrade the cluster to use Docker instead of containerd, generate a valid TLS certificate for your Kyma instance or [configure an external Docker registry](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-serverless/svls-07-set-external-registry/).
+
+- [Kubernetes](https://kubernetes.io/docs/setup/) (v1.19 or greater)
```
Comment (pbochynski) [inline docs/02-get-started/README.md:14]: Kyma doesn't work on Kubernetes 1.22
We need be more specific:  v1.19 - v1.21

**Review by pbochynski — CHANGES_REQUESTED** [review]

**Comment by mmitoraj** [comment]
PR blocked: installation of the Serverless component on AKS with k8s version 1.20.9 failed

**Review by mmitoraj — COMMENTED** [review]
The installation worked

**Review by pbochynski — DISMISSED** [review]

File: docs/04-operation-guides/operations/02-install-kyma.md [hunk 2]
```diff
@@ -8,9 +8,11 @@ Meet the prerequisites, provision a k3d cluster, and use the `deploy` command to
 
 ## Prerequisites
 
+>**CAUTION:** As of version 1.20, [Kubernetes deprecated Docker](https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/) as a container runtime in favor of [containerd](https://containerd.io/). Due to a different way in which containerd handles certificate authorities, Kyma's built-in Docker registry does not work correctly on clusters running with a self-signed TLS certificate on top of Kubernetes installation where containerd is used as a container runtime. If that is your case, either upgrade the cluster to use Docker instead of containerd, generate a valid TLS certificate for your Kyma instance or [configure an external Docker registry](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-serverless/svls-07-set-external-registry/).
+
+- [Kubernetes](https://kubernetes.io/docs/setup/) (v1.19 - recommended, 1.20, or 1.21)
+  - [k3d](https://k3d.io) (for the local installation only)
```
Comment (majakurcius) [inline docs/04-operation-guides/operations/02-install-kyma.md:14]:
```suggestion
  - [k3d](https://k3d.io) (for local installation only)
```
The process, not the instance.

File: docs/02-get-started/README.md [hunk 3]
```diff
@@ -9,9 +9,11 @@ All the steps are performed in the `default` Namespace.
 
 ## Prerequisites
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (1.19 or greater)
+>**CAUTION:** As of version 1.20, [Kubernetes deprecated Docker](https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/) as a container runtime in favor of [containerd](https://containerd.io/). Due to a different way in which containerd handles certificate authorities, Kyma's built-in Docker registry does not work correctly on clusters running with a self-signed TLS certificate on top of Kubernetes installation where containerd is used as a container runtime. If that is your case, either upgrade the cluster to use Docker instead of containerd, generate a valid TLS certificate for your Kyma instance or [configure an external Docker registry](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-serverless/svls-07-set-external-registry/).
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (v1.19 or higher)
 - [curl](https://github.com/curl/curl)
-- [k3d](https://k3d.io) (v4.0.0 or greater)
+- [k3d](https://k3d.io) (v4.0.0 or higher with [Kubernetes](https://kubernetes.io/docs/setup/) v1.19 - recommended, 1.20, or 1.21)
```
Comment (majakurcius) [inline docs/02-get-started/README.md:16]: Thank you for changing it to `higher`! 😻

File: docs/02-get-started/README.md [hunk 4]
```diff
@@ -9,9 +9,11 @@ All the steps are performed in the `default` Namespace.
 
 ## Prerequisites
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (1.19 or greater)
+>**CAUTION:** As of version 1.20, [Kubernetes deprecated Docker](https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/) as a container runtime in favor of [containerd](https://containerd.io/). Due to a different way in which containerd handles certificate authorities, Kyma's built-in Docker registry does not work correctly on clusters running with a self-signed TLS certificate on top of Kubernetes installation where containerd is used as a container runtime. If that is your case, either upgrade the cluster to use Docker instead of containerd, generate a valid TLS certificate for your Kyma instance or [configure an external Docker registry](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-serverless/svls-07-set-external-registry/).
+
```
Comment (majakurcius) [inline docs/02-get-started/README.md:13]:
```suggestion
```

I have doubts as to whether we should put this NOTE here. It's Get Started, it's supposed to be fairly concise and quick to do and it doesn't look great to require the user to do some switching before they even started installing Kyma. 
Maybe, for the sake of GSG, we can just use version 1.19 of Kubernetes here? The installation step links to the main Kyma installation doc anyway "for other ways to install Kyma", so the user will be able to see the CAUTION note there.

Optionally, we can link to that installation doc two lines below like this:
```
[k3d](https://k3d.io) (v4.0.0 or higher with [Kubernetes](https://kubernetes.io/docs/setup/) v1.19 - for higher versions, see [this tutorial](../../04-operation-guides/operations/02-install-kyma.md)
```

**Review by veichtj — APPROVED** [review]

**Review by majakurcius — APPROVED** [review]

**Pull request merged**